### ISSUE TITLE
Problem with get_reward function in SentiGAN.

### ISSUE THREAD

**Issue by DuanYiqun** [williamSYSU/TextGAN-PyTorch#24]
Dear author, thank you very much for your contribution. 
But I noticed that at real_data/SentiGAN_instructor.py line 143. 
`rewards = rollout_func.get_reward(target, cfg.rollout_num, self.dis)` see mingling haven't select different index in discriminator, where

>  reward 函数是这样     
def get_reward(self, sentences, rollout_num, dis, current_k=0):
        """
        get reward via Monte Carlo search
        :param sentences: size of batch_size * max_seq_len
        :param rollout_num:
        :param dis:
        :param current_k: current training gen
        :return: reward: [batch_size]
        """
        batch_size = sentences.size(0)
        rewards = torch.zeros([rollout_num * self.max_seq_len, batch_size]).float()
        if self.gpu:
            rewards = rewards.cuda()
        idx = 0
        for i in range(rollout_num):
            for given_num in range(1, self.max_seq_len + 1):
                samples = self.rollout_mc_search(sentences, given_num)
                out = dis.forward(samples)
                out = F.softmax(out, dim=-1)
                reward = out[:, current_k + 1]
                rewards[idx] = reward
                idx += 1
        rewards = torch.mean(rewards.view(batch_size, self.max_seq_len, rollout_num), dim=-1)
        return rewards

Seems like this reward doesn't include multi-class classification score?

**Comment by williamSYSU** [comment]
Thank you for your remind! 
Indeed, the generator should be specified when the discriminator calculates the reward. I have updated the codes with a parameter `current_k`, which is used for calculating rewards for a given generator.

**Comment by DuanYiqun** [comment]
Thank you for your response. 
However, do you find that, if set current_k with the reward function. 
It will meets a kind of mode collapse after the adversarial training?
For example: It will always be the same sentence: ' I love this movie xxx' something.

**Comment by williamSYSU** [comment]
Regardless of whether I add the parameter `current_k`, the mode collapse issue in SentiGAN is serious. During adversarial training, after only 10~20 training epochs, SentiGAN begins to generate repeated sentences.

**Comment by DuanYiqun** [comment]
Yeah，thank you for your information. I meet this problem, too.
Whether in this repo or other codes.

Any possible idea of this phenomenon?

**Comment by williamSYSU** [comment]
The mode collapse has been a serious issue in GAN for a long time, especially in text generation. The degree of mode collapse depends on the model, and I have no idea to alleviate this problem.
